### PR TITLE
Delete old avatars when a new one is approved

### PR DESCRIPTION
**Pull request by QuarpT**
Delete old avatars when a new one is approved. Part of GDPR requirements to prevent old avatars from being leaked

### CODE REVIEW

**Comment by nicl** [comment]
@QuarpT hey, sorry realise this is a while ago and you've probably forgotten. But just looking at this before I merge. What was the original motivation for moving stuff out into the separate and new 'AvatarUpdateService'? It seems like an unnecessary abstraction given Store already serves this kind of purpose.

File: api/src/test/scala/com/gu/adapters/http/AvatarServletTests.scala [hunk 1]
```diff
@@ -21,10 +21,12 @@ class AvatarServletTests extends TestHelpers {
   // See https://issues.scala-lang.org/browse/SI-5091
   val apiUrl: String = avatarServletProps.apiUrl
   val apiKey = avatarServletProps.apiKeys.head
+  val avatarStore = AvatarStore(new TestFileStore(storeProps.fsProcessedBucket), new TestKVStore(storeProps.kvTable), storeProps)
 
   addServlet(
     new AvatarServlet(
-      AvatarStore(new TestFileStore(storeProps.fsProcessedBucket), new TestKVStore(storeProps.kvTable), storeProps),
+      avatarStore,
+      //avatarUpdateService,
```
Comment (philmcmahon) [inline api/src/test/scala/com/gu/adapters/http/AvatarServletTests.scala]: what does this mean?

File: api/src/main/scala/com/gu/core/store/store.scala [hunk 2]
```diff
@@ -163,72 +195,99 @@ case class AvatarStore(fs: FileStore, kvs: KVStore, props: StoreProperties) exte
...
 
-    logIfError(s"Unable to update status for Avatar ID: $id. Avatar may be left in an inconsistent state.", result)
+    getActive(User(userId)).map(_.body) match {
+      case \/-(avatar) => delete(avatar)
+      case -\/(AvatarNotFound(_, _)) => AvatarNotDeleted(userId).right
+      case e: -\/[Error] => e
+    }
+  }
+
+  private[this] def updateKv(avatarId: String, status: Status, isActive: Boolean): \/[Error, Avatar] = {
```
Comment (philmcmahon) [inline api/src/main/scala/com/gu/core/store/store.scala]: what is a kv?

File: api/src/test/scala/com/gu/adapters/http/AvatarServletTests.scala (hunk 1 above)
Comment (nicl) [reply api/src/test/scala/com/gu/adapters/http/AvatarServletTests.scala]: Ah that's some code that got removed but incompletely! I'll remove it thanks.

File: api/src/main/scala/com/gu/core/store/store.scala (hunk 2 above)
Comment (nicl) [reply api/src/main/scala/com/gu/core/store/store.scala]: key value here

**Comment by nicl** [comment]
@philmcmahon thanks for the review. I've had to make a few more changes after testing - as noticed some bugs.

In general, the code here all feels a bit complex - I think this is mostly just the result of it being quite functional and using Scalaz's disjunction/either everywhere - so always up for suggestions on how to make it clearer!

It seems to work though :)

**Comment by philmcmahon** [comment]
:shipit:

**Comment by QuarpT** [comment]
@nicl only just saw your comment. I wanted to split independent functionality away from `store` which is accumulating a lot of business logic. I wouldn't call it an abstraction 😉

**Pull request merged**